controllers/storagecluster/cephblockpools.go

```diff
@@ -90,12 +91,21 @@ func (o *ocsCephBlockPools) reconcileCephBlockPool(r *StorageClusterReconciler,
 	_, err = ctrl.CreateOrUpdate(r.ctx, r.Client, cephBlockPool, func() error {
 		// Pass the poolSpec from the storageCluster CR
 
+		existingPoolSpec := cephBlockPool.Spec.PoolSpec
 		cephBlockPool.Spec.PoolSpec = storageCluster.Spec.ManagedResources.CephBlockPools.PoolSpec
 
 		// Set default values in the poolSpec as necessary
 		setDefaultDataPoolSpec(&cephBlockPool.Spec.PoolSpec, storageCluster)
 		cephBlockPool.Spec.PoolSpec.EnableRBDStats = true
 
+		// The bulk flag is set to true only during new pool creation, as setting it on existing pools can cause data movement.
```

Comment:
IMO, it would be helpful to add a note here that explains that this condition should only be possible when the CephBlockPool doesn't already exist, meaning this struct is initialized only in code.